src/core/address_space.cpp

```diff
@@ -245,21 +245,36 @@ struct AddressSpace::Impl {
     Impl() {
         // Allocate virtual address placeholder for our address space.
         void* hint_address = reinterpret_cast<void*>(SYSTEM_MANAGED_MIN);
+#ifdef __APPLE__
+        constexpr int virtual_flags = MAP_PRIVATE | MAP_ANONYMOUS | MAP_NORESERVE;
```

Comment:
If fixed mapping fails might be worth it doing something similar to what windows one. Depends on how much of the area macos process already has reserved though. Also consider disabling ASLR to make this deterministic

Reply:
I'll see if I can get a lower address to work.